### PR TITLE
extract mesh/line/pointCloud rendering from renderBufferDirect.

### PR DESCRIPTION
**Pull request by CodingFabian**
At the moment, renderBufferDirect is a large method body with a common
and 3 separate codes for each object type.
This patch extracts each code into its own method, this helps the
JavaScript JIT compiler inlining, but most importantly it allows easier
method level profiling. Right now a lot of profiles just show
`renderBufferDirect`.

### CODE REVIEW

**Comment by CodingFabian** [comment]
Easier review by ignoring whitespace: https://github.com/mrdoob/three.js/pull/6477/files?w=1

**Comment by mrdoob** [comment]
Could you remove the `Direct`? Just `renderMesh`, `renderLine` and `renderPointCloud`.
Also, where did you get that `?w=1` thing from? 😮

File: src/renderers/WebGLRenderer.js [hunk 1]
```diff
@@ -1010,380 +1010,393 @@ THREE.WebGLRenderer = function ( parameters ) {
...
+			_this.info.render.faces += position.array.length / ( 3 * position.itemSize );
 
-			// render particles
+		}
 
-			var mode = _gl.POINTS;
+	}
 
-			var index = geometry.attributes.index;
+	function renderPointCloudDirect ( camera, lights, fog, material, geometry, object ) {
```
Comment (mrdoob) [inline src/renderers/WebGLRenderer.js]: I think the parameters here don't match the other methods...? Have you tested this code?

Comment (CodingFabian) [inline src/renderers/WebGLRenderer.js]: sorry, my fault I did not save after last cleanup. fixed

**Comment by CodingFabian** [comment]
I did remove Direct and fixed my c&p issue with method signatures.
?w=1 is a eat trick to easily compare PRs which do a lot of whitespace changes due to method extraction/move

**Comment by mrdoob** [comment]
> ?w=1 is a eat trick to easily compare PRs which do a lot of whitespace changes due to method extraction/move

Yes, but where did you learnt about that trick?

**Comment by mrdoob** [comment]
Anyway. Looks good! Thanks! 😊

**Comment by CodingFabian** [comment]
I learned the trick when contributing to https://github.com/mozilla/pdf.js - guess its only passed on from dev to dev and not documented by github :)

**Comment by gero3** [comment]
https://github.com/blog/967-github-secrets is the only reference of such tricks that I know of.

**Comment by mrdoob** [comment]
Nice! Handy 😀

**Pull request merged**